### PR TITLE
MueLu: Adding new unit test

### PR DESCRIPTION
**Pull request by csiefer2**
Issue: #9200
Stakeholder feedback: n/a (testing only change)
Tests: Yes.  A new one.

### CODE REVIEW

File: packages/muelu/test/unit_tests/Hierarchy.cpp [hunk 1]
```diff
@@ -870,6 +870,112 @@ namespace MueLuTests {
...
+
+    // Multigrid Hierarchy
+    Hierarchy H(A);
+    H.setVerbLevel(Teuchos::VERB_HIGH);
+    H.SetMaxCoarseSize(10);
+
+
+
+
+    //  Hierarchy H(A);
```
Comment (GrahamBenHarper) [inline packages/muelu/test/unit_tests/Hierarchy.cpp:898]: This commented code should be removed

File: packages/muelu/test/unit_tests/Hierarchy.cpp [hunk 2]
```diff
@@ -870,6 +870,112 @@ namespace MueLuTests {
...
+    r = H.Setup(0, Teuchos::null,  rcpFromRef(M0), rcpFromRef(M1)); TEST_EQUALITY(r, false);
+    r = H.Setup(1, rcpFromRef(M0), rcpFromRef(M1), rcpFromRef(M2));   TEST_EQUALITY(r, false);
+    r = H.Setup(2, rcpFromRef(M1), rcpFromRef(M2), Teuchos::null ); TEST_EQUALITY(r, true);
+
+    RCP<Level> l0 = H.GetLevel(0);
+    RCP<Level> l1 = H.GetLevel(1);
+    RCP<Level> l2 = H.GetLevel(2);
+
+
+    /*
```
Comment (GrahamBenHarper) [inline packages/muelu/test/unit_tests/Hierarchy.cpp:933]: Same here

File: packages/muelu/test/unit_tests/Hierarchy.cpp [hunk 3]
```diff
@@ -870,6 +870,112 @@ namespace MueLuTests {
...
+    TEST_EQUALITY(l1->GetKeepFlag("P",            MueLu::NoFactory::get()), MueLu::Final);
+    TEST_EQUALITY(l1->GetKeepFlag("R",            MueLu::NoFactory::get()), MueLu::Final);
+    TEST_EQUALITY(l1->GetKeepFlag("PreSmoother",  MueLu::NoFactory::get()), MueLu::Final);
+    TEST_EQUALITY(l1->GetKeepFlag("PostSmoother", MueLu::NoFactory::get()), MueLu::Final);
+
+    TEST_EQUALITY(l2->GetKeepFlag("A",            MueLu::NoFactory::get()), MueLu::Final);
+    TEST_EQUALITY(l2->GetKeepFlag("P",            MueLu::NoFactory::get()), MueLu::Final);
+    TEST_EQUALITY(l2->GetKeepFlag("R",            MueLu::NoFactory::get()), MueLu::Final);
+    TEST_EQUALITY(l2->GetKeepFlag("PreSmoother",  MueLu::NoFactory::get()), MueLu::Final);
+    // TEST_EQUALITY(l2->GetKeepFlag("PostSmoother", MueLu::NoFactory::get()), MueLu::Final); // direct solve
```
Comment (GrahamBenHarper) [inline packages/muelu/test/unit_tests/Hierarchy.cpp:964]: and here

**Review by GrahamBenHarper — COMMENTED** [review]
Here's some feedback. I didn't see any serious issues

File: packages/muelu/test/unit_tests/Hierarchy.cpp (hunk 3 above)
Comment (csiefer2) [reply packages/muelu/test/unit_tests/Hierarchy.cpp:964]: Those are all commented out consistent the other tests.

**Review by cgcgcg — APPROVED** [review]

**Pull request merged**